cider/exec/nextgen/jitlib/llvmjit/LLVMJITUtils.h

```diff
@@ -0,0 +1,63 @@
+/*
+ * Copyright (c) 2022 Intel Corporation.
+ * Copyright (c) OmniSci, Inc. and its affiliates.
+ *
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+#ifndef JITLIB_LLVM_JIT_UTILS_H
+#define JITLIB_LLVM_JIT_UTILS_H
+
+#include "exec/nextgen/jitlib/base/ValueTypes.h"
+
+#include <llvm/IR/Constants.h>
+#include <llvm/IR/DerivedTypes.h>
+
+namespace jitlib {
+inline llvm::Type* getLLVMType(TypeTag tag, llvm::LLVMContext& ctx) {
```

Comment:
``` cpp
template<TypeTag tag>
inline constexpr llvm::Type* getLLVMType(llvm::LLVMContext& ctx) {
    if constexpr (tag == TypeTag::Void) {
        return llvm::Type::getVoidTy(ctx);
    } else if (...)
}
```
may be better?

Reply:
I implement this function as template initially. However, I found the actual ```TypeTag``` may be known at runtime mostly, so the template implementation is not available.

Reply:
Besides, the template parameter ```Type``` of ```Value``` may have the same problem.

Reply:
when you call `getLLVMType` , the type is known by compiler.